plugins/catalog-backend-module-geoserver-data/src/GeoserverDataProvider.ts

```diff
@@ -0,0 +1,314 @@
+          apiVersion: 'geoportia.se/v1alpha1',
+          kind: 'GeoserverStore',
+          metadata: {
+            name: dataStore.name,
+            namespace: workspace.name,
+            description: fullDataStoreObject?.dataStore?.description,
+            annotations: {
+              [ANNOTATION_LOCATION]: `url:${this.uri}`,
+              [ANNOTATION_ORIGIN_LOCATION]: `url:${this.uri}`,
+            },
+          },
+          spec: fullDataStoreObject?.dataStore || {},
+        };
+
+        entities.push(dataStoreEntity);
+      }
+
+      // Add coverage stores
+      const coverageStoresForWorkspaceObject =
+        await grc.datastores.getCoverageStores(workspace.name);
+      const coverageStoresForWorkspace =
+        coverageStoresForWorkspaceObject?.coverageStores?.coverageStore;
+      for (const coverageStore of coverageStoresForWorkspace || []) {
+        this.logger.debug(
+          `Found coverage store: ${coverageStore.name} at ${coverageStore.href} for workspace ${workspace.name}`,
+        );
+
+        const fullCoverageStoreObject = await grc.datastores.getCoverageStore(
+          workspace.name,
+          coverageStore.name,
+        );
+
+        const coverageStoreEntity: Entity = {
+          apiVersion: 'geoportia.se/v1alpha1',
+          kind: 'GeoserverStore',
+          metadata: {
+            name: coverageStore.name,
+            namespace: workspace.name,
+            description: fullCoverageStoreObject?.coverageStore?.description,
+            annotations: {
+              [ANNOTATION_LOCATION]: `url:${this.uri}`,
+              [ANNOTATION_ORIGIN_LOCATION]: `url:${this.uri}`,
+            },
+          },
+          spec: fullCoverageStoreObject?.coverageStore || {},
+        };
+        entities.push(coverageStoreEntity);
+      }
+
+      // Add WMS stores
+      const wmsStoresForWorkspaceObject = await grc.datastores.getWmsStores(
+        workspace.name,
+      );
+      const wmsStoresForWorkspace =
+        wmsStoresForWorkspaceObject?.wmsStores?.wmsStore;
+      for (const wmsStore of wmsStoresForWorkspace || []) {
+        this.logger.debug(
+          `Found WMS store: ${wmsStore.name} at ${wmsStore.href} for workspace ${workspace.name}`,
+        );
+
+        const fullWmsStoreObject = await grc.datastores.getWmsStore(
+          workspace.name,
+          wmsStore.name,
+        );
+
+        const wmsStoreEntity: Entity = {
+          apiVersion: 'geoportia.se/v1alpha1',
+          kind: 'GeoserverStore',
+          metadata: {
+            name: wmsStore.name,
+            namespace: workspace.name,
+            description: fullWmsStoreObject?.wmsStore?.description,
+            annotations: {
+              [ANNOTATION_LOCATION]: `url:${this.uri}`,
+              [ANNOTATION_ORIGIN_LOCATION]: `url:${this.uri}`,
+            },
+          },
+          spec: fullWmsStoreObject?.wmsStore || {},
+        };
+        entities.push(wmsStoreEntity);
+      }
+
+      // Add WMT stores
```

Comment:
```suggestion
      // Add WMTS stores
```

(same typo also present below)